### PR TITLE
new -compact mode, also use for eval when printing function objects

### PR DESCRIPTION
**Pull request by ldemailly**
```
$ ./grol -format -compact examples/sample.gr  2> /dev/null; echo
unless=macro(cond,iffalse,iftrue){quote(if !unquote(cond){unquote(iffalse)}else{unquote(iftrue)})}unless(10>5,print("BUG: not greater\n"),print("macro test: greater\n"))fact=func(n){log("called fact ",n)if n<=1{return 1}n*self(n-1)}a=[fact(5),"abc",76-3]m={"key":a,73:29}print("m is:",m,"\n")print("Outputting a smiley: 😀\n")first((m["key"]))
```

```
$ ./grol -format -compact examples/sample.gr | ./grol -format -
12:50:10.965 grol dev  go1.22.5 arm64 darwin - welcome!
12:50:10.965 grol dev  go1.22.5 arm64 darwin - welcome!
12:50:10.966 [INF] Formatting stdin
12:50:10.966 [INF] Formatting examples/sample.gr
12:50:10.966 [INF] All done
unless = macro(cond, iffalse, iftrue) {
	quote(if !unquote(cond) {
		unquote(iffalse)
	} else {
		unquote(iftrue)
	})
}
unless(10 > 5, print("BUG: not greater\n"), print("macro test: greater\n"))
fact = func(n) {
	log("called fact ", n)
	if n <= 1 {
		return 1
	}
	n * self(n - 1)
}
a = [fact(5), "abc", 76 - 3]
m = {"key":a, 73:29}
print("m is:", m, "\n")
print("Outputting a smiley: 😀\n")
first((m["key"]))
12:50:10.967 [INF] All done
```

relates a bit to #38

### CODE REVIEW

File: check_samples_double_format.sh [hunk 1]
```diff
@@ -8,5 +8,10 @@ for file in "$@"; do
     ./grol $file > /tmp/output1
     ./grol /tmp/format2 > /tmp/output2
     diff -u /tmp/output1 /tmp/output2
+    ./grol -format -compact $file > /tmp/format3
+    ./grol -format -compact /tmp/format3 > /tmp/format4
+    diff -u /tmp/format3 /tmp/format4
+    ./grol /tmp/format4 > /tmp/output3
+    diff -u /tmp/output1 /tmp/output3
```
Comment (ldemailly) [inline check_samples_double_format.sh:15]: this is the test, makes sure the compact form is stable and that it evaluates to same as non compact

probably could have a few unit test too

Comment (ldemailly) [reply check_samples_double_format.sh:15]: added unit test, not 100% sure there isn't a corner case but ... hopefully bug would get reported
